pages/movies/[id].js

```diff
@@ -0,0 +1,62 @@
+// import DetailList from "../../components/DetailList";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import Image from "next/image";
+import Link from "next/link";
+
+export default function MovieDetailPage({ movies }) {
+  console.log(movies);
+
+  const { query } = useRouter();
+  const { id } = query;
+
+  function getMovieById(id) {
+    return movies.find((movie) => movie.id === id);
+  }
+
+  const movieDetail = getMovieById(id);
+
+  if (!movieDetail) return <h2>loading..</h2>;
+
+  return (
+    <main>
+      <h1>{movieDetail.title}</h1>
+      <MovieImageContainer>
+        <Image
+          src={movieDetail.picture}
+          width={384}
+          height={259.2}
```

Comment:
That's a rather specific number. If you actually need that number, ignore my comment. Otherwise you may want to check if rounding the number down makes any difference.